src/Aspire.Dashboard/ResourceService/DashboardClient.cs

```diff
@@ -168,7 +168,23 @@ X509CertificateCollection GetFileCertificate()
                 var filePath = _dashboardOptions.ResourceServiceClient.ClientCertificate.FilePath;
                 var password = _dashboardOptions.ResourceServiceClient.ClientCertificate.Password;
 
-                return [new X509Certificate2(filePath, password)];
+                var certBytes = File.ReadAllBytes(filePath);
+
+                var certContentType = X509Certificate2.GetCertContentType(filePath);
```

Comment:
Why are you reading the file twice? You could use the overload taking `ReadOnlySpan<Byte>` and pass certBytes instead of filePath...

Reply:
Thanks yes, that was the intention. Good spot. Will fix before merging.